simpledotorg/simple-server
# Export disaggregated HTN data to Bangladesh DHIS2

**Story card:** [sc-8304](https://app.shortcut.com/simpledotorg/story/8304)

## Because

Bangladesh wants disaggregated hypertension data to be sent to their DHIS2 instance. The queries added by this PR can also be used when generating reports that are not necessarily needed by the dashboard.

## This addresses

- Refactors existing DHIS2 exporters by pulling out common features to a separate file
- Adds patient states queries to get the required hypertension data, disaggregate data by gender and age. It adds the following queries:
  - Cumulative registrations
  - Monthly registrations
  - Dead patients
  - Lost to follow up patients
  - Cumulative assigned patients
  - Controlled patients
  - Uncontrolled patients
  - Missed visits

## Test instructions

- Run the queries and compare them against corresponding reporting facility states columns.
- Run the exporter rake task to send disaggregated data to DHIS2 sandbox and ensure correctness.
